app/services/builders/project.rb

```diff
@@ -11,12 +11,14 @@ def call
     private
 
     def fetch_or_create
-      ::Project.find_or_create_by!(github_id: @repository_data['id']) do |project|
-        project.name = @repository_data['name']
-        project.description = @repository_data['description']
-        project.is_private = @repository_data['private']
-        project.relevance = ::Project.relevances[:ignored] if @repository_data['archived']
-      end
+      project = ::Project.find_or_initialize_by(github_id: @repository_data['id'])
+      project.name = @repository_data['name']
+      project.description = @repository_data['description']
+      project.is_private = @repository_data['private']
+      project.relevance = ::Project.relevances[:ignored] if @repository_data['archived']
+      project.save!
+
+      project
```

Comment:
I believe we can use `.tap`.
`::Project.find_or_initialize_by(github_id: @repository_data['id']).tap |project|`
It not relevant anyway.

Reply:
Absolutely! Nice catch!